lib/puppet/catalog-diff/compilecatalog.rb

```diff
@@ -97,14 +97,14 @@ def compile_catalog(node_name, server, certless)
       end
 
       uri = URI("https://#{server}:#{port}#{endpoint}")
-      Puppet.debug("Connecting to server: #{uri}")
+      ssl_context = Puppet::CatalogDiff::Tlsfactory.ssl_context(tls_cert, tls_key, tls_ca)
       begin
         ret = if certless
-                Puppet.runtime[:http].post(uri, body.to_json, headers: headers)
+                Puppet.runtime[:http].post(uri, body.to_json, headers: headers, options: { ssl_context: ssl_context })
               else
-                Puppet.runtime[:http].get(uri, headers: headers)
+                Puppet.runtime[:http].get(uri, headers: headers, options: { ssl_context: ssl_context })
               end
-        raise "HTTP request to PuppetDB failed with: HTTP #{ret.code} - #{ret.reason}" unless ret.success?
+        raise "HTTP request to Puppetserver #{server} failed with: HTTP #{ret.code} - #{ret.reason}" unless ret.success?
```

Comment:
this was a typo previously. the request was always going to the puppetserver, never puppetdb